### PR TITLE
[Maintain][Manifest] add elementwise unary-direct entries

### LINKED ISSUE #1073: [Maintain][Manifest] add elementwise unary-direct entries to ops_manifest.yaml
### Symptom / Motivation

The `elementwise` family currently has **no entries** in `tileops/ops_manifest.yaml`. This issue covers **24 unary ops with a direct `torch.<op>` reference** (one of four sub-issues splitting the elementwise family by sub-table; sibling issues cover unary functional / binary / multi-input).

This issue records the input table; the agent loops `/add-manifest` over the table.

### Root Cause Analysis

N/A — family-wide spec creation. The op classes were added to code without corresponding manifest entries.

### Related Files

- `tileops/ops_manifest.yaml`
- `tileops/ops/elementwise.py`

## Goal

Bring all 24 unary-direct elementwise ops under manifest tracking at `status: spec-only`. Single batch PR.

## Plan

**Plan type: fixed**

Op list (the agent should embed this verbatim as the operative artefact; one row per op):

### Unary — direct `torch.<op>`

| op_name | ref_api | ref_url |
|---|---|---|
| ExpFwdOp | torch.exp | https://docs.pytorch.org/docs/stable/generated/torch.exp.html |
| LogFwdOp | torch.log | https://docs.pytorch.org/docs/stable/generated/torch.log.html |
| SqrtFwdOp | torch.sqrt | https://docs.pytorch.org/docs/stable/generated/torch.sqrt.html |
| RsqrtFwdOp | torch.rsqrt | https://docs.pytorch.org/docs/stable/generated/torch.rsqrt.html |
| AbsFwdOp | torch.abs | https://docs.pytorch.org/docs/stable/generated/torch.abs.html |
| NegFwdOp | torch.neg | https://docs.pytorch.org/docs/stable/generated/torch.neg.html |
| ReciprocalFwdOp | torch.reciprocal | https://docs.pytorch.org/docs/stable/generated/torch.reciprocal.html |
| SignFwdOp | torch.sign | https://docs.pytorch.org/docs/stable/generated/torch.sign.html |
| SinFwdOp | torch.sin | https://docs.pytorch.org/docs/stable/generated/torch.sin.html |
| CosFwdOp | torch.cos | https://docs.pytorch.org/docs/stable/generated/torch.cos.html |
| FloorFwdOp | torch.floor | https://docs.pytorch.org/docs/stable/generated/torch.floor.html |
| CeilFwdOp | torch.ceil | https://docs.pytorch.org/docs/stable/generated/torch.ceil.html |
| RoundFwdOp | torch.round | https://docs.pytorch.org/docs/stable/generated/torch.round.html |
| TruncFwdOp | torch.trunc | https://docs.pytorch.org/docs/stable/generated/torch.trunc.html |
| ErfFwdOp | torch.erf | https://docs.pytorch.org/docs/stable/generated/torch.erf.html |
| Log1pFwdOp | torch.log1p | https://docs.pytorch.org/docs/stable/generated/torch.log1p.html |
| Expm1FwdOp | torch.expm1 | https://docs.pytorch.org/docs/stable/generated/torch.expm1.html |
| SigmoidFwdOp | torch.sigmoid | https://docs.pytorch.org/docs/stable/generated/torch.sigmoid.html |
| TanhFwdOp | torch.tanh | https://docs.pytorch.org/docs/stable/generated/torch.tanh.html |
| LogicalNotFwdOp | torch.logical_not | https://docs.pytorch.org/docs/stable/generated/torch.logical_not.html |
| BitwiseNotFwdOp | torch.bitwise_not | https://docs.pytorch.org/docs/stable/generated/torch.bitwise_not.html |
| IsnanFwdOp | torch.isnan | https://docs.pytorch.org/docs/stable/generated/torch.isnan.html |
| IsinfFwdOp | torch.isinf | https://docs.pytorch.org/docs/stable/generated/torch.isinf.html |
| IsfiniteFwdOp | torch.isfinite | https://docs.pytorch.org/docs/stable/generated/torch.isfinite.html |

## Constraints

- Single batch PR for these 24 ops. Title `[Maintain][Manifest] add elementwise unary-direct entries`.
- Each op invocation is `/add-manifest <op_name> <ref_url>`. Skill is idempotent and one-entry-per-invocation; loop in the agent driver, not in the skill. The driver MUST stay on a single branch and accumulate edits — do not let the skill open per-op PRs.
- Do NOT touch op / kernel / test / bench code. Manifest-only PR.
- All entries land at `status: spec-only` — `align-op` does the flip later.
- Sibling sub-issues cover the other 38 PyTorch-referenced ops (unary functional / binary / multi-input). 5 composite ops without a direct PyTorch reference (SiluAndMulFwdOp, GeluAndMulFwdOp, GeluTanhAndMulFwdOp, AlibiFwdOp, SinusoidalFwdOp) are tracked separately as design follow-ups.

## Acceptance Criteria

- [ ] AC-1: The resulting PR contains exactly 24 new manifest entries (one per row).
- [ ] AC-2: Validator (`python scripts/validate_manifest.py --check-op <op_name>`) passes L0 on every added op.
- [ ] AC-3: For any op where `/add-manifest` returns BLOCKED, the agent records the BLOCKED case in the PR body and excludes that op from the batch — leaves it for follow-up. The PR ships with whatever subset succeeded.

### PR DESCRIPTION
**Pull request by lcy-seso**
## Summary

Bring all 24 unary-direct elementwise ops under manifest tracking at `status: spec-only`. This is a manifest-only PR — no op/kernel/test/bench code is touched. Per the manifest trust model, alignment of impl-vs-spec is deferred to follow-up PRs.

Ops added (24): `ExpFwdOp`, `LogFwdOp`, `SqrtFwdOp`, `RsqrtFwdOp`, `AbsFwdOp`, `NegFwdOp`, `ReciprocalFwdOp`, `SignFwdOp`, `SinFwdOp`, `CosFwdOp`, `FloorFwdOp`, `CeilFwdOp`, `RoundFwdOp`, `TruncFwdOp`, `ErfFwdOp`, `Log1pFwdOp`, `Expm1FwdOp`, `SigmoidFwdOp`, `TanhFwdOp`, `LogicalNotFwdOp`, `BitwiseNotFwdOp`, `IsnanFwdOp`, `IsinfFwdOp`, `IsfiniteFwdOp`.

BLOCKED ops: none (all 24 added). See `.foundry/runs/issue-1073/blocked.md` for a note on `RoundFwdOp` L1 signature caveat (manifest matches `torch.round` public API; impl alignment is a follow-up).

Closes #1073

## Test plan

- [x] **AC-1**: PR contains exactly 24 new manifest entries (24 minus 0 BLOCKED). Evidence: `grep -E "^  [A-Z][A-Za-z0-9]*FwdOp:" tileops/ops_manifest.yaml` shows all 24 expected ops; `git show 399b9e1 --stat` reports 1 file changed, 730 insertions(+).
- [x] **AC-2**: `python scripts/validate_manifest.py --check-op <op_name> --levels=schema` passes L0 on every added op. Full-manifest L0 (`--levels=schema`) also passes. Note: full `--check-op` on `RoundFwdOp` reports 1 L1 [signature] error (manifest `decimals` param absent in impl); handled per trust model via `status: spec-only`.
- [x] **AC-3**: For any op where `/add-manifest` returned BLOCKED, the agent records BLOCKED and excludes that op. 0 BLOCKED ops; `.foundry/runs/issue-1073/blocked.md` documents this with the RoundFwdOp L1 caveat.
- [x] Validator suite green: 170/170 validator tests pass.

### CODE REVIEW

File: tileops/ops_manifest.yaml [hunk 1]
```diff
@@ -2471,3 +2471,733 @@ ops:
...
+        - "y.shape == x.shape"
+
+    workloads:
+      - {x_shape: [4096, 4096], dtypes: [float16, bfloat16, float32], label: "elementwise-16M"}
+      - {x_shape: [16384, 16384], dtypes: [float16, bfloat16], label: "elementwise-256M"}
+
+    roofline:
+      vars:
+        N_total: "product(x.shape)"
+      flops: "N_total"
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply tileops/ops_manifest.yaml]: Addressed in 0f6c6b0.

Evidence:
- Current `TanhFwdOp.roofline.flops` is `5 * N_total` (matches your suggestion):
```yaml
# tileops/ops_manifest.yaml line 3041-3042
# tanh(x) = 2 * sigmoid(2x) - 1: ~5 ops/elem (1 mul for 2x + 4 sigmoid ops)
flops: "5 * N_total"
```
- Commit 0f6c6b0 bumped flops from `N_total` → `5 * N_total` and added the analytical-derivation comment so the model is consistent with SigmoidFwdOp's `4 * N_total`.

Current HEAD: 0f6c6b0

Comment (Ibuki-wind) [inline tileops/ops_manifest.yaml]: `TanhFwdOp.roofline.flops` models tanh as `N_total`, but the same manifest models `SigmoidFwdOp` as `4 * N_total`; using one FLOP for tanh makes the SOL denominator inconsistent and will understate the analytical cost for this activation. → Set this to an implementation-consistent formula such as `5 * N_total` and add a short derivation comment matching the style above.

**Review by Ibuki-wind — CHANGES_REQUESTED** [review]
### New issues
- `tileops/ops_manifest.yaml:3041` — `TanhFwdOp` roofline undercounts tanh relative to the manifest's sigmoid model.

### Overall
REQUEST_CHANGES: one manifest roofline formula needs correction before this can be approved.
Schema-only validation of the PR manifest passed; no CI checks were reported.

File: tileops/ops_manifest.yaml (hunk 1 above)
Comment (lcy-seso) [reply tileops/ops_manifest.yaml]: Adopted. Set `TanhFwdOp.roofline.flops` to `5 * N_total` with analytical comment `# tanh(x) = 2 * sigmoid(2x) - 1: ~5 ops/elem (1 mul for 2x + 4 sigmoid ops)` matching SigmoidFwdOp's style. See 0f6c6b0.

**Review by Ibuki-wind — APPROVED** [review]
### Prior blockers
- **resolved** — `tileops/ops_manifest.yaml:3041` `TanhFwdOp.roofline.flops` now uses `5 * N_total` with a derivation comment.

### Overall
Prior blocker is resolved and the incremental manifest-only change is clean.
No CI checks were reported for this round.
